### PR TITLE
Quote identifiers when building Pinot query

### LINKED ISSUE #9373: date column name not parsing properly in passthrough query from trino to pinot
Hello all,
I have connected trino with pinot. Its working fine but the only problem that i am facing in tables whose columns have “date” as the column name for datefield.

I am getting the error 

> Error = org.apache.calcite.sql.parser.SqlParseException: Encountered "date ," at line 1, column 8.

PFA the file for reference.  I humbly request you to please suggest some solutions on this

[pinot_trino_error.txt](https://github.com/trinodb/trino/files/7225421/pinot_trino_error.txt)

### PR DESCRIPTION
**Pull request by ebyhr**
Fixes #9373

### CODE REVIEW

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/DynamicTablePqlExtractor.java [hunk 1]
```diff
@@ -113,4 +116,9 @@ public static String encloseInParentheses(String value)
     {
         return format("(%s)", value);
     }
+
+    private static String quoteIdentifier(String identifier)
+    {
+        return format("\"%s\"", identifier);
```
Comment (findepi) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/DynamicTablePqlExtractor.java:122]: `checkArgument` that identifier doesn't contain `"` (or escape it properly for Pinot)

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/FilterToPinotSqlConverter.java [hunk 2]
```diff
@@ -201,6 +201,6 @@ private String formatLiteral(Literal literal)
     private String formatIdentifier(Identifier identifier)
     {
         PinotColumnHandle pinotColumnHandle = (PinotColumnHandle) requireNonNull(columnHandles.get(identifier.getName()), "Column not found");
-        return pinotColumnHandle.getColumnName();
+        return "\"" + pinotColumnHandle.getColumnName() + "\"";
```
Comment (findepi) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/FilterToPinotSqlConverter.java:204]: column name might contain `"`

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/PinotQueryBuilder.java [hunk 3]
```diff
@@ -53,19 +53,19 @@ public static String generatePql(PinotTableHandle tableHandle, List<PinotColumnH
...
         if (columnHandles.isEmpty()) {
             // This occurs when the query is SELECT COUNT(*) FROM pinotTable ...
-            columnNames = ImmutableList.of("*");
+            quotedColumnNames = ImmutableList.of("*");
         }
         else {
-            columnNames = columnHandles.stream()
-                    .map(PinotColumnHandle::getColumnName)
+            quotedColumnNames = columnHandles.stream()
+                    .map(column -> "\"" + column.getColumnName() + "\"")
```
Comment (findepi) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/PinotQueryBuilder.java:63]: column name might contain `"`

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/PinotQueryBuilder.java [hunk 4]
```diff
@@ -169,12 +169,12 @@ private static String toConjunct(String columnName, String operator, Object valu
         if (value instanceof Slice) {
             value = ((Slice) value).toStringUtf8();
         }
-        return format("%s %s %s", columnName, operator, singleQuote(value));
+        return format("\"%s\" %s %s", columnName, operator, singleQuote(value));
```
Comment (findepi) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/PinotQueryBuilder.java:172]: column name might contain `"`

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/PinotQueryBuilder.java [hunk 5]
```diff
@@ -169,12 +169,12 @@ private static String toConjunct(String columnName, String operator, Object valu
...
     }
 
     private static String inClauseValues(String columnName, List<Object> singleValues)
     {
-        return format("%s IN (%s)", columnName, singleValues.stream()
+        return format("\"%s\" IN (%s)", columnName, singleValues.stream()
```
Comment (findepi) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/PinotQueryBuilder.java:177]: column name might contain `"`

**Review by findepi — APPROVED** [review]

**Comment by ebyhr** [comment]
@elonazoulay  Handling escape in WHERE condition in Pinot looks weird. Do you have any idea to fix it? Tested in Pinot query console:
```sql
SELECT "esc""ape", "esc""ape" = 'test_escape' FROM reserved_keyword 
```

esc"ape | equals(esc"ape,'test_escape')
-- | --
test_escape | 1
test_escape | 1

```sql
SELECT "esc""ape", "esc""ape" = 'test_escape' FROM reserved_keyword 
WHERE "esc""ape" = 'test_escape'
```
→ empty

**Comment by elonazoulay** [comment]
> @elonazoulay Handling escape in WHERE condition in Pinot looks having silent correctness issue. Is it a known bug in the old version? 0.8.0 is fine as far as I confirmed. Test via Pinot query console:
> 
> ```sql
> SELECT "esc""ape", "esc""ape" = 'test_escape' FROM reserved_keyword 
> ```
> 
> esc"ape	equals(esc"ape,'test_escape')
> test_escape	1
> test_escape	1
> ```sql
> SELECT "esc""ape", "esc""ape" = 'test_escape' FROM reserved_keyword 
> WHERE "esc""ape" = 'test_escape'
> ```
> 
> → empty

Quoting identifiers was not implemented at all before you added the functionality. Once I saw your commit I added it to all places, select, filter, group by (and having - but in a subsequent pr). Also added a test for aliases including keywords. lmk if there's anything else you see that needs to be done.

**Pull request merged**